0-bootstrap/modules/jenkins-agent/main.tf

```diff
@@ -95,10 +95,11 @@ resource "google_compute_instance" "jenkins_agent_gce_instance" {
   service_account {
     email = google_service_account.jenkins_agent_gce_sa.email
     scopes = [
-      // TODO(caleonardo): These scopes will need to change.
+      // TODO(caleonardo): These scopes should be reduced and avoid using "cloud-platform".
       "https://www.googleapis.com/auth/compute.readonly",
-      "https://www.googleapis.com/auth/devstorage.read_only",
+      "https://www.googleapis.com/auth/devstorage.read_write",
```

Comment:
If you have cloud-platform, you probably don't need this scope anymore. Also, given the number of APIs this service account will be hitting, it may be an ongoing game of whack-a-mole trying to reduce this in future. I agree with @mikelaramie 's sentiment around focusing on the IAM permissions in this case

Reply:
Thank you @rjerrems - If the GCE Agent will impersonate the Terraform seed SA to make the changes, can the Agent have reduced scopes? - if so, I think we should reduce them.  

If the Agent needs all the scopes anyways (I mean, even using the impersonation), then yes, we can just use `cloud-platform`.

Reply:
I think we will need to test this, but my assumption would be that even with impersonation all of the scopes would be required. Happy to reduce if we can

Reply:
It's in our [best practices](https://cloud.google.com/compute/docs/access/create-enable-service-accounts-for-instances#best_practices) to allow `cloud-platform` and rely on IAM to restrict what a Service Account can or can't do.  Scopes can provide a false sense of security, because they only apply to the GCE instance making the API call.

Reply:
Thank you for the details @mikelaramie - This is now only using scope `cloud-platform`.